sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -498,6 +498,12 @@ object JdbcUtils extends Logging with SQLConfHelper {
         }
       }
 
+    case TimestampNTZType if metadata.contains("logical_time_type") =>
+      (rs: ResultSet, row: InternalRow, pos: Int) =>
+        val micros = nullSafeConvert[java.sql.Time](rs.getTime(pos + 1), t =>
+          localDateTimeToMicros(LocalDateTime.of(LocalDate.EPOCH, t.toLocalTime)))
+        row.update(pos, micros)
```

Comment:
Instead of here, shall we move these line before of the existing ` case TimestampNTZType =>`? At line 510 (old line number)?